src/bsoncxx/decimal128.hpp

```diff
@@ -0,0 +1,98 @@
+// Copyright 2016 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#include <bsoncxx/stdx/string_view.hpp>
+
+#include <bsoncxx/config/prelude.hpp>
+
+namespace bsoncxx {
+BSONCXX_INLINE_NAMESPACE_BEGIN
+
+///
+/// Represents an IEEE 754-2008 BSON Decimal128 value in a platform-independent way.
+///
+class BSONCXX_API decimal128 {
+   public:
+    ///
+    /// Constructs a BSON Decimal128 value representing zero.
+    ///
+    decimal128() = default;
+
+    ///
+    /// Constructs a BSON Decimal128 from high and low 64-bit big-endian parts.
+    ///
+    /// @param high are the high 64-bits.
```

Comment:
I guess the rendering is more like a DT/DD pair.  So name followed by some sentence like thing is fine.  I personally tend to dislike extremely short sentence fragments with periods.